Don't commit for read-only query

uranusjr opened the pull request.

uranusjr committed:
Don't commit for read-only query

ephraimbuddy commented on task_sdk/src/airflow/sdk/definitions/asset/__init__.py:
```diff
@@ -429,10 +429,13 @@ class AssetAlias(BaseAsset):
-        with create_session() as session:
+        # We don't use create_session or provide_session here because this is
+        # called in the scheduler when commit is prohibited (for HA reasons).
+        # Those functions assume you want to commit the result on exit.
```
Isn't it possible to pass the scheduler session to this?

uranusjr replied:
It is, but this function is also (indirectly) used by places where there’s not an active session. I feel it’s easier to open a new one for this (it doesn’t really cost much).

uranusjr replied:
I changed this to accept an optional session when it makes sense instead.

uranusjr committed:
Pass session from outside when we can

This does not use the create_session/provide_session/NEW_SESSION
paradigm because it requires importing them globally, which is not
allowed in the SDK. We also do not want to create a session unless
absolutely needed.
